ruma-api-macros/src/api/response.rs

```diff
@@ -65,7 +65,7 @@ impl Response {
                 }
                 ResponseField::NewtypeRawBody(_) => {
                     quote_spanned! {span=>
-                        #field_name: response.into_body()
+                        #field_name: resp.body().to_vec()
```

Comment:
I was wondering whether there would be a case like this. This now being a full copy of the response makes my whole idea moot unfortunately¹. I don't want to have worse runtime performance for somewhat nicer generated code.

I have another idea though: We could hide the repetitive

```rust
match <expr> {
    Ok(val) => val,
    Err(e) => return <error type>::new(<actual error>, <reponse / request>).into(),
}
```

behind a helper function (or two, one for requests and one for responses). I hope that would make the macro code easier to read.

Does that makes sense to you? Do you have other ideas for how to make the code more readable?

¹: If you want a more in-depth explanation of why I don't think this can work (without `try` blocks 😄), ping me on Matrix.